solutions/silver/usaco-416.mdx

```diff
@@ -7,8 +7,94 @@ author: Brad Ma, Kevin Sheng
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_lazy_silver.html)
 
-## Implementation
+### Implementation (Prefix Sums)
 
+**Time Complexity** $\mathcal{O}(N ^ 2)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <fstream>
+#include <vector>
+
+using std::endl;
+using std::vector;
+using std::max;
+
+int main() {
+	std::ifstream read("lazy.in");
+
+	int n;
+	int k;
+	read >> n >> k;
+	// the side length needed to accomodate the 45 degree rotation
+	int new_n = 2 * n - 1;
+	vector<vector<int>> field(new_n, vector<int>(new_n));
```

Comment:
ngl don't think this is sufficiently different from the og sol to warrant a diff section?
by "prefix sums" i was thinking storing pref sums for every row and iterating through them for each query

Reply:
elaborate?